skl2onnx/operator_converters/TextVectorizer.py

```diff
@@ -46,16 +75,43 @@ def convert_sklearn_text_vectorizer(scope, operator, container):
             "only stip_accents=None is supported.")
 
     options = container.get_options(
-            op, dict(sep=[' ', '.', '?', ',', ';', ':', '!']))
-    if set(options) != {'sep'}:
+            op, dict(sep="DEFAULT",
+                     regex=None))
+    if set(options) != {'sep', 'regex'}:
         raise RuntimeError("Unknown option {} for {}".format(
                                 set(options) - {'sep'}, type(op)))
-    default_pattern = '(?u)\\b\\w\\w+\\b'
-    default_separators = options['sep']
-    if op.token_pattern != default_pattern:
-        raise NotImplementedError(
-            "Only the default tokenizer based on default regular expression "
-            "'{0}' is implemented.".format(default_pattern))
+
+    if op.analyzer == 'word':
+        default_pattern = '(?u)\\b\\w\\w+\\b'
+        if options['sep'] == "DEFAULT" and options['regex'] is None:
+            warnings.warn("Converter for TfidfVectorizer will use scikit-learn regular expression by default in 0.16",
+                          DeprecationWarning)
+            default_separators = [' ', '.', '?', ',', ';', ':', '!']
+            regex = None
+            if op.token_pattern != default_pattern:
+                raise NotImplementedError(
+                    "Only the default tokenizer based on default regular expression "
+                    "'{0}' is implemented.".format(default_pattern))
+        elif options['regex'] is not None:
+            if options['regex']:
+                regex = options['regex']
+            else:
+                regex = op.token_pattern
+                if regex == default_pattern:
+                    regex = '\\\\b(\\\\w\\\\w+)\\\\b'
```

Comment:
Using \\ instead of \ is unexpected but otherwise onnxruntime fails. It needs to be investigated.

Reply:
Fixed.